### PR TITLE
Ngwmn 2018 - USGS Site Metadata can be populated via NWIS

### PR DESCRIPTION
**Pull request by mbucknell**
Before making a pull request
----------------------------

- [X] Make sure all tests run
- [X] Update the changelog appropriately

Title
-----------
USGS Site Metadata can be populated via NWIS

Description
-----------
I added a custom view and form to the registry admin which will allow the user to enter a site no and the application will fetch the site from NWIS and populated a MonitoringLocation with the meta data. If this is successfully the user will be redirected to the editing page.

Added a button on the admin monitoringlocation page to go to this view. That button will only be visible for USGS users.

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

**Comment by jkreft-usgs** [comment]
does it deal with more than one site being returned per site number?

**Comment by mbucknell** [comment]
We use both the entered site number and default the agency_cd to 'USGS' so it will only get one.

File: wellregistry/registry/admin.py [hunk 1]
```diff
@@ -27,40 +38,31 @@ def clean(self):
 
         if cleaned_data.get('well_depth') and not cleaned_data.get('well_depth_units'):
             raise forms.ValidationError(
-                 'If Well depth is populated, then you must enter a Well depth unit')
+                'If Well depth is populated, then you must enter a Well depth unit')
```
Comment (mbucknell) [inline wellregistry/registry/admin.py:41]: These changes in the clean method are just reformatting. pylint was flagging a bunch of these. I really recommend that make runlint is run before submitting a PR. I found lots of things in the changes that I made to that needed to be improved.

File: wellregistry/registry/migrations/0013_registry_update_well_depth.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# The below suppresses pylint message: Module name "*" doesn't conform to snake_case naming style
+# pylint: disable-msg=C0103
+# Enable check for the rest of the file
+# pylint: enable-msg=C0103
+"""
+# Updating fields including well depth which did not allow for well depths over 100)
```
Comment (mbucknell) [inline wellregistry/registry/migrations/0013_registry_update_well_depth.py:6]: I did have to make a change to the well depth field but most of these are left over from the PR for validations. Recommend that make runmigrations is run before making a PR to make sure that we don't get migration warnings about missing migrations.

File: wellregistry/registry/models.py [hunk 3]
```diff
@@ -183,7 +181,7 @@ class MonitoringLocation(models.Model):
     alt_method = models.CharField(max_length=300, blank=True, verbose_name='Altitude method')
     alt_acy = models.CharField(max_length=300, blank=True, verbose_name='Altitude accuracy')
 
-    well_depth = models.DecimalField(max_digits=11, decimal_places=8, null=True, blank=True)
+    well_depth = models.DecimalField(max_digits=11, decimal_places=7, null=True)
```
Comment (mbucknell) [inline wellregistry/registry/models.py:184]: This field was not configure to hold values > 999 so I modified it. I looked in the legacy database and no well depths are larger than 9999.

**Review by orosz-usgs — APPROVED** [review]
Looks good.

File: wellregistry/registry/admin.py (hunk 1 above)
Comment (ajmccarthy) [reply wellregistry/registry/admin.py:41]: Thanks for the comment.  I'll keep that in mind!

File: wellregistry/registry/models.py (hunk 3 above)
Comment (ajmccarthy) [reply wellregistry/registry/models.py:184]: blank = True might need to be added back here? Well depth can be null in some situations.

**Review by ajmccarthy — COMMENTED** [review]
An exciting change.  Thanks for including me as a reviewer.  Possibly well_depth in the MonitoringLocation class needs to have blank= True.

File: wellregistry/registry/models.py (hunk 3 above)
Comment (mbucknell) [reply wellregistry/registry/models.py:184]: It's a decimal field so you don't need blank=True and null=True

Comment (mbucknell) [reply wellregistry/registry/models.py:184]: But now that I think about it, if we don't set blank=True then the field is required in the form. I will revert this change will doing the current ticket. Thanks for being so detail oriented on this!

**Pull request merged**